R/redcap-event-read.R

```diff
@@ -0,0 +1,193 @@
+#' @title
+#' Export Events
+#'
+#' @description
+#' Export Events of a REDCap project
+#'
+#' @param redcap_uri The
+#' [uri](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)/url
+#' of the REDCap server
+#' typically formatted as "https://server.org/apps/redcap/api/".
+#' Required.
+#' @param token The user-specific string that serves as the password for a
+#' project.  Required.
+#' @param verbose A boolean value indicating if `message`s should be printed
+#' to the R console during the operation.  The verbose output might contain
+#' sensitive information (*e.g.* PHI), so turn this off if the output might
+#' be visible somewhere public. Optional.
+#' @param config_options A list of options passed to [httr::POST()].
+#' See details at [httr::httr_options()]. Optional.
+#' @param handle_httr The value passed to the `handle` parameter of
+#' [httr::POST()].
+#' This is useful for only unconventional authentication approaches.  It
+#' should be `NULL` for most institutions.  Optional.
+#'
+#' @return
+#' Currently, a list is returned with the following elements:
+#' * `has_events`: a `logical` value indicating if the REDCap project has
+#' events (*i.e.*, "TRUE") or is a classic non-longitudinal project
+#' (*i.e.*, "FALSE").
+#' * `data`: a [tibble::tibble()] with one row per arm-event combination.  The
+#' columns are `event_name` (a human-friendly string), `arm_num` (an integer),
+#' `unique_event_name` (a string), `custom_event_label` (a string), and
+#' `event_id` (a double).
+#' * `success`: A boolean value indicating if the operation was apparently
+#' successful.
+#' * `status_code`: The
+#' [http status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
+#' of the operation.
+#' * `outcome_message`: A human readable string indicating the operation's
+#' outcome.
+#' * `elapsed_seconds`: The duration of the function.
+#' * `raw_text`: If an operation is NOT successful, the text returned by
+#' REDCap.  If an operation is successful, the `raw_text` is returned as an
+#' empty string to save RAM.
+#'
+#' @author
+#' Will Beasley, Ezra Porter
```

Comment:
@ezraporter, you did this.  I'm kidding.  But seriously, at least put yourself first.